### PR TITLE
Feature.pwm

### PR DESCRIPTION
**Pull request by bjornborg**

### CODE REVIEW

File: code/proxy-miniature/gpio/src/Gpio.cpp [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    return direction;
+  } else {
+    cerr << "[" << getName() << "] Could not open " << gpioDirectionFilename 
+        << "." << std::endl;
+    gpioDirectionFile.close();
+    return "";
+  }
+}
+
+void Gpio::SetValue(uint16_t const a_pin, bool const val)
```
Comment (olbender) [inline code/proxy-miniature/gpio/src/Gpio.cpp]: Missing "a_"

File: resources/odvd/ODVDMiniature.odvd [hunk 2]
```diff
@@ -16,7 +16,22 @@
...
+}
+
+message opendlv.proxy.DigitalReading [id = 159] {
+  uint16 pin [id = 1];
+  int32 value [id = 2];
+}
+
+message opendlv.proxy.PwmRequest [id = 155] {
+  uint16 pin [id = 1];
+  uint32 dutyCycle [id = 2];
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: Since non-standard unit: dutyCycleNs

File: resources/odvd/ODVDMiniature.odvd [hunk 3]
```diff
@@ -16,7 +16,22 @@
...
+}
+
 message opendlv.proxy.AnalogReading [id = 173] {
   uint16 pin [id = 1];
-  float val [id = 2];
+  float value [id = 2];
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: Change to 'voltage'.

File: resources/odvd/ODVDMiniature.odvd [hunk 4]
```diff
@@ -16,7 +16,22 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+message opendlv.proxy.DigitalRequest [id = 170] {
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: Change to: ToggleRequest

File: resources/odvd/ODVDMiniature.odvd [hunk 5]
```diff
@@ -16,7 +16,22 @@
...
+  uint16 pin [id = 1];
+  int32 value [id = 2];
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: change to:

enum ToggleState {
        Off = 0,
        On = 1
    };
ToggleState state [id = 1];

File: resources/odvd/ODVDMiniature.odvd [hunk 6]
```diff
@@ -16,7 +16,22 @@
...
+}
+
+message opendlv.proxy.DigitalReading [id = 159] {
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: Change to ToggleReading

File: resources/odvd/ODVDMiniature.odvd [hunk 7]
```diff
@@ -16,7 +16,22 @@
...
+  uint16 pin [id = 1];
+  int32 value [id = 2];
```
Comment (olbender) [inline resources/odvd/ODVDMiniature.odvd]: enum ToggleState {
        Off = 0,
        On = 1
    };
ToggleState state [id = 1];

File: usecases/latest/host.tme290/configuration [hunk 8]
```diff
@@ -70,5 +70,26 @@ odsupercomponent.pulsetimeack.exclude = odcockpit
...
+proxy-miniature-gpio.systemPath = /sys/class/gpio
+proxy-miniature-gpio.pins = 30,31
+proxy-miniature-gpio.values = 0,1
+proxy-miniature-gpio.directions = in,out
+
+proxy-miniature-pwm.debug = 1
+proxy-miniature-pwm.systemPath = /sys/class/pwm/pwmchip2
+proxy-miniature-pwm.pins = 0,1
+proxy-miniature-pwm.periods = 20000000,20000000
+proxy-miniature-pwm.dutyCycles = 1500000,1500000
```
Comment (olbender) [inline usecases/latest/host.tme290/configuration]: Change to periodsNs and dutyCyclesNs

**Review by olbender — CHANGES_REQUESTED** [review]
Code is very nice, but some name changes

**Comment by bjornborg** [comment]
Please, review again.

**Review by olbender — APPROVED** [review]

**Pull request merged**